signer/src/eth.rs

```diff
@@ -0,0 +1,346 @@
+        }
+        /// Equivalent to `{DEV_PHRASE}//Bob`.
+        pub fn bob() -> Keypair {
+            Keypair::from_uri(&SecretUri::from_str("//Bob").unwrap()).unwrap()
+        }
+        /// Equivalent to `{DEV_PHRASE}//Charlie`.
+        pub fn charlie() -> Keypair {
+            Keypair::from_uri(&SecretUri::from_str("//Charlie").unwrap()).unwrap()
+        }
+        /// Equivalent to `{DEV_PHRASE}//Dave`.
+        pub fn dave() -> Keypair {
+            Keypair::from_uri(&SecretUri::from_str("//Dave").unwrap()).unwrap()
+        }
+        /// Equivalent to `{DEV_PHRASE}//Eve`.
+        pub fn eve() -> Keypair {
+            Keypair::from_uri(&SecretUri::from_str("//Eve").unwrap()).unwrap()
+        }
+        /// Equivalent to `{DEV_PHRASE}//Ferdie`.
+        pub fn ferdie() -> Keypair {
+            Keypair::from_uri(&SecretUri::from_str("//Ferdie").unwrap()).unwrap()
+        }
+        /// Equivalent to `{DEV_PHRASE}//One`.
+        pub fn one() -> Keypair {
+            Keypair::from_uri(&SecretUri::from_str("//One").unwrap()).unwrap()
+        }
+        /// Equivalent to `{DEV_PHRASE}//Two`.
+        pub fn two() -> Keypair {
+            Keypair::from_uri(&SecretUri::from_str("//Two").unwrap()).unwrap()
+        }
+    }
+}
+
+#[cfg(feature = "subxt")]
+mod subxt_compat {
+    use super::*;
+
+    impl<T: subxt::Config> subxt::tx::Signer<T> for Keypair
+    where
+        T::AccountId: From<AccountId20>,
+        T::Address: From<AccountId20>,
+        T::Signature: From<Signature>,
+    {
+        fn account_id(&self) -> T::AccountId {
+            self.account_id().into()
+        }
+
+        fn address(&self) -> T::Address {
+            self.account_id().into()
+        }
+
+        fn sign(&self, signer_payload: &[u8]) -> T::Signature {
+            self.sign(signer_payload).into()
+        }
+    }
+}
+
+#[cfg(test)]
+mod test {
+    use super::*;
+    use proptest::prelude::*;
+
+    use hex::ToHex;
+    enum StubEthRuntimeConfig {}
+
+    impl subxt::Config for StubEthRuntimeConfig {
+        type Hash = subxt::utils::H256;
+        type AccountId = super::AccountId20;
+        type Address = super::AccountId20;
+        type Signature = super::Signature;
+        type Hasher = subxt::config::substrate::BlakeTwo256;
+        type Header =
+            subxt::config::substrate::SubstrateHeader<u32, subxt::config::substrate::BlakeTwo256>;
+        type ExtrinsicParams = subxt::config::SubstrateExtrinsicParams<Self>;
+        type AssetId = u32;
+    }
+
+    type Signer = dyn subxt::tx::Signer<StubEthRuntimeConfig>;
+
+    prop_compose! {
+        fn keypair()(seed in any::<[u8; 32]>()) -> Keypair {
+            let secret = SecretKey::from_slice(&seed).expect("valid secret key");
+            let inner = secp256k1::Keypair::from_secret_key(
+                &Secp256k1::new(),
+                &secret,
+            );
+
+            Keypair(ecdsa::Keypair(inner))
+        }
+    }
+
+    proptest! {
```

Comment:
Cool! I haven't used proptest before so this is neat to see! I wonder whether it's worth opening another issue to add proptest tests for the other signers too?

Reply:
https://github.com/paritytech/subxt/issues/1506